Compiler error: A function can not runned directly from a map

### Describe the bug

When you try to run a function added to a map, V will show a compiler error. 

### Expected Behavior

Print `0` to console

### Current Behavior

A compiler error
```
error: too many arguments provided to function-like macro invocation
        int b = ((int (*) ())(*(voidptr*)map_get(ADDR(map, ADDR(map, a), &(string[]){_SLIT("a")}, &(voidptr[]){ new_map(sizeof(string), sizeof(anon_fn___int), &map_hash_string, &map_eq_string, &map_clone_string, &map_free_string) })))), &(string[]){_SLIT("a")}, &(voidptr[]){ 0 })))();

                   ^
/data/data/com.termux/files/usr/tmp/v_10013/bug.2890048671409307881.tmp.c:181:9: note: macro 'ADDR' defined here
#define ADDR(type, expr) (&((type[]){expr}[0]))
        ^
/data/data/com.termux/files/usr/tmp/v_10013/bug.2890048671409307881.tmp.c:12421:43: error: use of undeclared identifier 'ADDR'
        int b = ((int (*) ())(*(voidptr*)map_get(ADDR(map, ADDR(map, a), &(string[]){_SLIT("a")}, &(voidptr[]){ new_map(sizeof(string), sizeof(anon_fn___int), &map_hash_string, &map_eq_string, &map_clone_string, &map_free_string) })))), &(string[]){_SLIT("a")}, &(voidptr[]){ 0 })))();
                                                 ^
/data/data/com.termux/files/usr/tmp/v_10013/bug.2890048671409307881.tmp.c:12421:231: error: expected identifier or '('
        int b = ((int (*) ())(*(voidptr*)map_get(ADDR(map, ADDR(map, a), &(string[]){_SLIT("a")}, &(voidptr[]){ new_map(sizeof(string), sizeof(anon_fn___int), &map_hash_string, &map_eq_string, &map_clone_string, &map_free_string) })))), &(string[]){_SLIT("a")}, &(voidptr[]){ 0 })))();




                     ^
```

### Reproduction Steps

```
mut a := map[string]map[string]fn()int
a["a"]["a"] = fn ()int{return 0}
b := a["a"]["a"]()
println(b)
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.3.3 f9c186a

### Environment details (OS name and version, etc.)

V full version: V 0.3.3 f9c186a
OS: termux, 4.9.227-24346293, #1 SMP PREEMPT Fri Apr 7 14:05:38 +07 2023
Processor: 8 cpus, 32bit, little endian

getwd: /data/data/com.termux/files/home
vexe: /data/data/com.termux/files/home/app/v/v
vexe mtime: 2023-04-11 13:54:00

vroot: OK, value: /data/data/com.termux/files/home/app/v
VMODULES: OK, value: /data/data/com.termux/files/home/.vmodules
VTMP: OK, value: /data/data/com.termux/files/usr/tmp/v_10013

Git version: git version 2.40.0
Git vroot status: 0.3.4-27-gb50dac5e
.git/config present: true

CC version: clang version 16.0.1
thirdparty/tcc status: thirdparty-unknown-unknown de82a130

--- a/vlib/v/gen/c/index.v
+++ b/vlib/v/gen/c/index.v
@@ -474,6 +474,7 @@ fn (mut g Gen) index_of_map(node ast.IndexExpr, sym ast.TypeSymbol) {
 		}
 		tmp_opt := if gen_or { g.new_tmp_var() } else { '' }
 		tmp_opt_ptr := if gen_or { g.new_tmp_var() } else { '' }
+		mut is_fn_last_index_call := false
 		if gen_or {
 			g.write('${elem_type_str}* ${tmp_opt_ptr} = (${elem_type_str}*)(map_get_check(')
 		} else {
@@ -482,6 +483,8 @@ fn (mut g Gen) index_of_map(node ast.IndexExpr, sym ast.TypeSymbol) {
 					g.write('((')
 					g.write_fn_ptr_decl(&elem_sym.info, '')
 					g.write(')(*(voidptr*)map_get(')
+					is_fn_last_index_call = true
+					g.is_fn_index_call = false
 				}
 			} else {
 				g.write('(*(${elem_type_str}*)map_get(')
@@ -506,7 +509,7 @@ fn (mut g Gen) index_of_map(node ast.IndexExpr, sym ast.TypeSymbol) {
 		g.write('}')
 		if gen_or {
 			g.write('))')
-		} else if g.is_fn_index_call {
+		} else if is_fn_last_index_call {
 			g.write(', &(voidptr[]){ ${zero} })))')
 		} else {
 			g.write(', &(${elem_type_str}[]){ ${zero} }))')

--- a/vlib/v/tests/nested_map_of_fn_call_test.v
+++ b/vlib/v/tests/nested_map_of_fn_call_test.v
@@ -1,0 +1,9 @@
+fn test_nested_map_of_fn_call() {
+	mut a := map[string]map[string]fn () int{}
+	a['a']['a'] = fn () int {
+		return 0
+	}
+	b := a['a']['a']()
+	println(b)
+	assert b == 0
+}

